### PR TITLE
Support ChainerX in wavenet example

### PR DESCRIPTION
**Pull request by niboshi**
Part of #6661

### CODE REVIEW

File: examples/wavenet/generate.py [hunk 1]
```diff
@@ -58,11 +64,10 @@
     args.model, decoder, 'updater/model:main/predictor/decoder/')
 
 # Non-autoregressive generate
-if use_gpu:
-    x = chainer.cuda.to_gpu(x, device=args.gpu)
-    condition = chainer.cuda.to_gpu(condition, device=args.gpu)
-    encoder.to_gpu(device=args.gpu)
-    decoder.to_gpu(device=args.gpu)
+x = device.send(x)
+condition.device.send(x)
```
Comment (hvy) [inline examples/wavenet/generate.py]: Isn't `condition` a `numpy.ndarray` array here?

File: examples/wavenet/train.py [hunk 2]
```diff
@@ -94,15 +104,16 @@
...
 trainer = chainer.training.Trainer(
     updater, (args.epoch, 'epoch'), out=args.out)
 
 # Extensions
 snapshot_interval = (args.snapshot_interval, 'iteration')
 display_interval = (args.display_interval, 'iteration')
-trainer.extend(extensions.Evaluator(valid_iter, model, device=args.gpu))
-trainer.extend(extensions.dump_graph('main/loss'))
+trainer.extend(extensions.Evaluator(valid_iter, model, device=device))
+if device.xp is not chainerx:
```
Comment (hvy) [inline examples/wavenet/train.py]: A minor but should we add a TODO similar to https://github.com/chainer/chainer/pull/6739/files#diff-30d5da2c4d62b2408b6ff6087fdd3ef3R104 ?

File: examples/wavenet/generate.py (hunk 1 above)
Comment (niboshi) [reply examples/wavenet/generate.py]: Sorry, it seems I forgot to push the latest revision.

File: examples/wavenet/train.py (hunk 2 above)
Comment (niboshi) [reply examples/wavenet/train.py]: Ditto. Please review once again.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
CI failure seems irrelevant. LGTM!

**Pull request merged**